src/Datadog.Trace/Logging/LibLogScopeEventSubscriber.cs

```diff
@@ -209,5 +284,38 @@ private void SetSerilogCompatibleLogContext(string service, string version, stri
                 RemoveAllCorrelationIdentifierContexts();
             }
         }
+
+#if NETFRAMEWORK
+#pragma warning disable SA1202 // Elements must be ordered by access
+#pragma warning disable SA1204 // Static elements must appear before instance elements
+        private static void RefreshIISPreAppState(ulong traceId)
+        {
+            Debug.Assert(_executingIISPreStartInit, $"{nameof(_executingIISPreStartInit)} should always be true when entering {nameof(RefreshIISPreAppState)}");
+
+            if (_performAppDomainFlagChecks)
+            {
+                object state = AppDomain.CurrentDomain.GetData(NamedSlotName);
+                _executingIISPreStartInit = state is bool boolState && boolState;
+            }
+            else
+            {
+                var stackTrace = new StackTrace(false);
+                var initialStackFrame = stackTrace.GetFrame(stackTrace.FrameCount - 1);
+                var initialMethod = initialStackFrame.GetMethod();
+
+                _executingIISPreStartInit = initialMethod.DeclaringType.FullName.Equals("System.Web.Hosting.HostingEnvironment", StringComparison.OrdinalIgnoreCase)
```

Comment:
After the value transitions back to false, can't you unsubscribe the `RefreshIISPreAppState` handler?

Reply:
Yes you're right. I'll add that

Reply:
@kevingosse do you know if there are any caveats of removing the event handler while the same event handler is still running?

Reply:
Done in aacd0a06d838ffbde0cac80270ca73335b6b0f0f